Kernel/PerformanceEventBuffer.cpp

```diff
@@ -107,11 +110,37 @@ KResult PerformanceEventBuffer::append_with_eip_and_ebp(ProcessID pid, ThreadID
         break;
     case PERF_EVENT_THREAD_EXIT:
         break;
+    case PERF_EVENT_CONTEXT_SWITCH:
+        event.data.context_switch.next_pid = arg1;
+        event.data.context_switch.next_tid = arg2;
+        break;
+    case PERF_EVENT_KMALLOC:
+        event.data.kmalloc.size = arg1;
+        event.data.kmalloc.ptr = arg2;
+        break;
+    case PERF_EVENT_KFREE:
+        event.data.kfree.size = arg1;
+        event.data.kfree.ptr = arg2;
+        break;
+    case PERF_EVENT_PAGE_FAULT:
+        break;
     default:
         return EINVAL;
     }
 
-    auto backtrace = raw_backtrace(ebp, eip);
+    decltype(raw_backtrace(ebp, eip)) backtrace;
+
+    {
+        InterruptDisabler disabler;
```

Comment:
With the changes to `enter_profiler` I don't think you need to disable interrupts at all.